mitodl/micromasters
# Fixed dashboard API course status regression

#### What are the relevant tickets?

Fixes #1697

#### What's this PR do?

Fixes the dashboard API regression that prevents past course runs from being evaluated

#### Where should the reviewer start?

dashboard/views.py and dashboard/api.py

#### How should this be manually tested?

Make sure you have the standard seed data loaded via the management command. Run the following management commands to get a course with past runs:

```
manage.py alter_data --action=set_course_to_failed --username=YOUR_USERNAME --program-title='Analog' --course-title='100'
manage.py alter_data --action=add_past_failed_run --username=YOUR_USERNAME --program-title='Analog' --course-title='100' --add-if-exists
```

You can also use `--email=` instead of `--username=` to match your desired user. You can run `alter_data --help` and `alter_data --usage` for more details.

Check the dashboard for the 'Analog Learning' program and make sure it's showing you that the 100-level course is failed with another past failed run before it.
